src/main/java/org/folio/rest/impl/DataExportHelper.java

```diff
@@ -28,15 +28,19 @@
 
   private Future<String> getDownloadFileName(String jobExecutionId, String exportFileId, String tenantId) {
     return jobExecutionService.getById(jobExecutionId, tenantId)
-      .map(job -> job.map(jb -> jb.getExportedFiles()
+      .map(job -> job.getExportedFiles()
         .stream()
         .filter(expFile -> expFile.getFileId()
           .equals(exportFileId))
-        .findFirst())
+        .findFirst()
         .orElseThrow(() -> new ServiceException(HttpStatus.HTTP_NOT_FOUND, String.format("Job with id: %s not found", jobExecutionId))))
-      .map(exportedFile -> exportedFile.map(ExportedFile::getFileName)
-          .orElseThrow(() -> new ServiceException(HttpStatus.HTTP_NOT_FOUND, String.format("Export File with id: %s not found: ", exportFileId))));
-
+      .compose(exportedFile -> {
```

Comment:
I know the above line was not introduced in this PR, but I think if the job is not found then it will be a failed future thrown by getById method, it doesn't come to line 36. So , the error message should belong to file  not found